src/rgw/rgw_rest_s3.cc

```diff
@@ -2541,20 +2541,30 @@ void RGWCompleteMultipart_ObjStore_S3::send_response()
   if (op_ret == 0) { 
     dump_start(s);
     s->formatter->open_object_section_in_ns("CompleteMultipartUploadResult", XMLNS_AWS_S3);
-    if (!s->bucket_tenant.empty()) {
-      if (s->info.domain.length()) {
-        s->formatter->dump_format("Location", "%s.%s.%s",
-          s->bucket_name.c_str(),
-          s->bucket_tenant.c_str(),
-          s->info.domain.c_str());
+    std::string domain = (!s->info.domain.empty()) ? s->info.domain :
+      [&]() -> std::string {
+      std::string domain = "http://";
```

Comment:
Shouldn't we consider HTTPS here?

Reply:
Yes.  I thought of that after typing.  And we make this same mistake elsewhere.  But are we supposed to just pick http/s, or match the incoming proto?

Reply:
We should stick to the incoming proto, I bet.